src/chain/TxChainBuilder.js

```diff
@@ -128,6 +128,14 @@ class TxChainBuilderImpl
         return this.source.getUtxo(id)
     }
 
+    /**
+     * @param {TxOutputId} id
+     * @returns {Promise<boolean>}
+     */
+    async hasUtxo(id) {
+        return !!(await this.getUtxo(id))
```

Comment:
I think `getUtxo` will fail instead of returning `undefined` or `null` if the UTxO doesn't exist
 
I suggest the following implementation:
```
async hasUtxo(id) {
        for (let i = 0; i < this.txs.length; i++) {
            const tx = this.txs[i]
            if (tx.id().isEqual(id.txId)) {
                return id.index < tx.body.outputs.length
            }
        }

        return this.source.hasUtxo(id)
}
```